### ISSUE TITLE
Restore signed packages on Big Endian CPUs

### ISSUE THREAD

**Issue by nealef** [NuGet/Home#9547]
## Details about Problem

NuGet.exe - NuGet version (5.5.0.6319)

## Detailed repro steps so we can see the same problem

1. Run NuGet install of any package - for example build NuGet.Client

On Big Endian platforms:
```
/home/neale/NuGet.Client/build/bootstrap.proj : error : One or more errors occurred. (Byte signature not found in package archive:  0x06054B50)
```
In ```EndOfCentralDirectoryRecord.cs``` the value ```Signature``` is defined as:
```
internal const uint Signature = 0x06054b50;
```
And is (only?) used to check/locate the end of central directory signature via:
```
var byteSignature = BitConverter.GetBytes(Signature);
```
Which results in different values depending on platform-endianess. 

- LE: ```504b0506```
- BE: ```06054b50```

As ```Signature``` can't be declared as ```const byte[]``` would this suffice?
```
diff --git a/src/NuGet.Core/NuGet.Packaging/Signing/Archive/EndOfCentralDirectoryRecord.cs b/src/NuGet.Core/NuGet.Packaging/Signing/Archive/EndOfCentralDirectoryRecord.cs
index f754422a3..086a76860 100644
--- a/src/NuGet.Core/NuGet.Packaging/Signing/Archive/EndOfCentralDirectoryRecord.cs
+++ b/src/NuGet.Core/NuGet.Packaging/Signing/Archive/EndOfCentralDirectoryRecord.cs
@@ -52,6 +52,9 @@ private static void SeekToEndOfCentralDirectoryRecord(BinaryReader reader)
             var byteSignature = BitConverter.GetBytes(Signature);
             var length = reader.BaseStream.Length;

+           if (!BitConverter.IsLittleEndian)
+                Array.Reverse(byteSignature);
+
             if (length < byteSignature.Length)
             {
                 ThrowByteSignatureNotFoundException(byteSignature);
```

**Comment by donnie-msft** [comment]
@heng-liu

**Comment by heng-liu** [comment]
Hi @nealef , may I know what kind of hardware/OS are you using? Thanks!

**Comment by nealef** [comment]
Mono 6.x on Linux on s390x (aka IBM Z aka "the Mainframe").


-------- Original message --------

**Comment by nealef** [comment]
Let me know if you want me to test any binaries.

**Comment by nealef** [comment]
Hi,
 Is there any progress with this issue? My offer of big-endian resources stands. We have a Jenkins client ready and waiting if that is useful.

Neale

**Comment by zivkan** [comment]
G'day, mate. I was back home in December, which feels like such a long time ago now.

I applied your patch and built locally: [NuGet.zip](https://github.com/NuGet/Home/files/4880579/NuGet.zip)

I'm also asking around internally to see what big-endian hardware we might have access to. It's unclear to me whether .NET Core is officially supported on big-endian hardware, or if any testing and fixes come through customer reports and pull requests. I have two raspberry pis at home, but apparently they support both big and little endian, and most operating systems (including all offially supported OSes) configure little-endian mode, so it's not all that easy for me to get access to big endian hardware to test on.

**Comment by nealef** [comment]
Thanks. I can give you access to a Big Endian virtual machine. Just send me your public key. It will be a CentOS 7 system that has mono 6.13 installed. 

Thanks for the zip! I will give it a test drive.

**Comment by nealef** [comment]
It seems to do the trick but I'll give it a more thorough workout tomorrow. Many thanks!

**Comment by zivkan** [comment]
Being the idealist that I am, I refactored the code a little, as you can see in the diff in the PR I just created.  If you'd like to test the binary with the new changes, it's here: [NuGet.zip](https://github.com/NuGet/Home/files/4885168/NuGet.zip).  Again, this was built locally, so missing authenticode signatures and does not correspond to any official, supported release. Assuming the PR gets merged quickly, it will be part of NuGet 5.8, probably preview 1.

Also, I found out that Microsoft does not directly/officially support big endian compatibility for either Mono or .NET Core. It's up to the community to test, report issues, and preferably, provide PRs. Thank you for your offer to give access to a machine to test with, but as we don't support big endian hardware ourselves, we rely on community feedback.

**Comment by nealef** [comment]
@zivkan This appears to work as well. Thanks.
